### PR TITLE
Issue#64 Add set and setIn operations (based on existing merge)

### PR DESCRIPTION
**Pull request by haggholm**
Add simple implementation of set and setIn by constructing a path and invoking merge, which already does all the hard stuff, checking, managing mutability, &c.

### CODE REVIEW

**Comment by haggholm** [comment]
Refers to #64 (apparently the commit comment didn’t do the trick, sorry).

I’m not sure whether adding methods like this is really the right thing to do, but I followed to the model of `merge`. Perhaps it would be better to follow a `lodash`-like model of exported functions to avoid conflicts with named properties.

File: test/ImmutableObject/test-set.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    var s = JSC.string()().replace(/[^\w]/g, '_');
+    return /^\d+$/.test(s) ? s + 'a' : s;
+  }
+
+  describe("#set", function() {
+    it("sets a property by name", function () {
+      check(100, [TestUtils.ComplexObjectSpecifier()], function(ob) {
+        var clone = _.cloneDeep(ob),
+          prop = getPathComponent(),
+          value = Math.random();
```
Comment (rtfeldman) [inline test/ImmutableObject/test-set.js]: Any reason not to use `JSC` for this?

File: src/seamless-immutable.js [hunk 2]
```diff
@@ -343,6 +360,8 @@
     addPropertyTo(obj, "without", without);
     addPropertyTo(obj, "asMutable", asMutableObject);
     addPropertyTo(obj, "instantiateEmptyObject", instantiateEmptyObject);
+    addPropertyTo(obj, "set", set);
+    addPropertyTo(obj, "setIn", setIn);
```
Comment (rtfeldman) [inline src/seamless-immutable.js]: Mind documenting these in `README`?

**Comment by rtfeldman** [comment]
Generally looks good! Sorry I took so long reviewing it.

I left a couple of comments, and there's also a merge conflict to resolve.

**Comment by haggholm** [comment]
Sorry I took even longer getting back! Added to README, changed a few random values to use JSC, not sure if that’s precisely what you meant: haven’t used JSC before (though if it’s inspired by QuickCheck, I expect to do so again).

**Comment by johnste** [comment]
Could this also be made to work with arrays as well? Currently it's cumbersome to update an item in an array.

**Comment by haggholm** [comment]
It can be done with arrays; I added it to arrays. I also found some weird edge cases where the object versions didn't work properly (with nested paths); I didn't nail it down, as my main concern was making the tests pass for my added methods, but wonkiness seemed to ensue sometimes when e.g. merging a complex object with a `{foo: NaN}` value. The new version is a bit more straightforward (IMO) and doesn't use `merge()`, so it's not affected, but it could use more digging. I can't take more company time for it right now, though!

**Comment by rtfeldman** [comment]
Thanks for this! :tada:
